src/idl_gen_cpp.cpp

```diff
@@ -1637,6 +1654,10 @@ class CppGenerator : public BaseGenerator {
     code_ += "};";  // End of table.
     code_ += "";
 
+    if (parser_.opts.generate_object_based_api) {
```

Comment:
why is this moved down from its original location?

Reply:
This was moved due to the sizeof() in the operator new, so that the full definition is available.

Reply:
Can you instead move the operator new further down?